### PR TITLE
Added sys_analyzer_linux.py

### PR DESCRIPTION
**Pull request by dmitryermilov**
The tool checks health of media stack on Linux

Signed-off-by: Dmitry Ermilov <dmitry.ermilov@intel.com>

### CODE REVIEW

File: tools/sys_analyzer/sys_analyzer_linux.py [hunk 1]
```diff
@@ -0,0 +1,704 @@
...
+Abstract: Script for checking the environment for Media SDK,
+SDK for OpenCL Applications install and run readiness
+
+Notes: Run ./sys_analyzer_linux.py or ./sys_analyzer_linux.py -v
+
+"""
+
+import os, sys, platform
+import os.path
+import pkgconfig # To install: "python3 -m pip install pkgconfig" or "pip3 install pkgconfig"
```
Comment (dvrogozh) [inline tools/sys_analyzer/sys_analyzer_linux.py]: Please, add `tools/sys_analyzer/README.md` and describe:
1. Prerequisites for the script
2. Which checks are performed by the script

Also, add a note about the script to top-level README.md

File: tools/sys_analyzer/sys_analyzer_linux.py [hunk 2]
```diff
@@ -0,0 +1,704 @@
...
+        print_info("i915 driver in use by Intel video adapter",loglevelcode.INFO)
+    else:
+        print_info("Intel video adapter not using i915",loglevelcode.ERROR)
+
+    check_libdrm()
+
+    print("--------------------------")
+    print("Component Smoke Tests:")
+    print("--------------------------")
+    check_msdk_api()
```
Comment (dvrogozh) [inline tools/sys_analyzer/sys_analyzer_linux.py]: I wish to see successful output of the script....

Can you add a check whether HuC is available on the system w/ a note which features might be missing if it is not available?

Comment (dmitryermilov) [reply tools/sys_analyzer/sys_analyzer_linux.py]: > I wish to see successful output of the script....

From SKL:
```
./sys_analyzer_linux.py
--------------------------
Hardware readiness checks:
--------------------------
 [ OK ] Processor name: Intel(R) Core(TM) i5-6260U CPU @ 1.80GHz
--------------------------
OS readiness checks:
--------------------------
 [ OK ] GPU visible to OS
--------------------------
User and kernel mode drivers health checks:
--------------------------
 [ ERROR ] user not in the 'video' group. Add with 'usermod -a -G video {user}'
 [ OK ] libva pkg-config found
 [ OK ] vainfo reports valid codec entry points
 [ OK ] /dev/dri/renderD128 connects to Intel i915
--------------------------
Component Smoke Tests:
--------------------------
 [ OK ] Media SDK HW API level:1.30

```

```
./sys_analyzer_linux.py -v
--------------------------
Hardware readiness checks:
--------------------------
 [ OK ] Processor name: Intel(R) Core(TM) i5-6260U CPU @ 1.80GHz
 [ INFO ] Intel Processor
 [ INFO ] Processor brand: Core
--------------------------
OS readiness checks:
--------------------------
 [ INFO ] GPU PCI id     : 1926
 [ INFO ] GPU description: SKL ULT GT3
 [ OK ] GPU visible to OS
 [ INFO ] no nomodeset in GRUB cmdline (good)
 [ INFO ] Linux distro   : CentOS Linux 7.5.1804
 [ INFO ] Linux kernel   : 5.0.3-1.el7.elrepo.x86_64
 [ INFO ] glibc version  : 2.17
 [ INFO ] gcc version    : 4.8.5-28) (>=4.8.2 suggested)
--------------------------
User and kernel mode drivers health checks:
--------------------------
 [ ERROR ] user not in the 'video' group. Add with 'usermod -a -G video {user}'
 [ OK ] libva pkg-config found
 [ INFO ] Intel iHD used by libva
 [ OK ] vainfo reports valid codec entry points
 [ INFO ] i915 driver in use by Intel video adapter
 [ OK ] /dev/dri/renderD128 connects to Intel i915
--------------------------
Component Smoke Tests:
--------------------------
 [ OK ] Media SDK HW API level:1.30
 [ INFO ] no OpenCL include files. Are Intel components installed?

```
From ICL:
```
./sys_analyzer_linux.py -v
--------------------------
Hardware readiness checks:
--------------------------
 [ OK ] Processor name: Genuine Intel(R) CPU 0000 @ 1.00GHz
 [ INFO ] Intel Processor
 [ INFO ] Processor not Xeon or Core.
--------------------------
OS readiness checks:
--------------------------
 [ INFO ] GPU PCI id     : 8A52
 [ INFO ] GPU description: ICL GT2
 [ OK ] GPU visible to OS
 [ INFO ] no nomodeset in GRUB cmdline (good)
 [ INFO ] Linux distro   : Ubuntu 18.04
 [ INFO ] Linux kernel   : 5.3.0
 [ INFO ] glibc version  : 2.27
 [ INFO ] gcc version    : 7.3.0 (>=4.8.2 suggested)
--------------------------
User and kernel mode drivers health checks:
--------------------------
 [ OK ] user in 'video' group
 [ INFO ] libva package not found in default PKG_CONFIG locations. Trying again with PKG_CONFIG_PATH=/opt/intel/mediasdk/lib64/pkgconfig/
 [ INFO ] Trying again with PKG_CONFIG_PATH=/opt/intel/mediasdk/lib64/pkgconfig/
 [ OK ] libva found with PKG_CONFIG_PATH=/opt/intel/mediasdk/lib64/pkgconfig/
 [ INFO ] Intel iHD used by libva
 [ OK ] vainfo reports valid codec entry points
 [ INFO ] i915 driver in use by Intel video adapter
 [ OK ] /dev/dri/renderD128 connects to Intel i915
--------------------------
Component Smoke Tests:
--------------------------
 [ OK ] Media SDK HW API level:1.30
 [ INFO ] no OpenCL include files. Are Intel components installed?

```

File: tools/sys_analyzer/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    --------------------------
+    OS readiness checks:
+    --------------------------
+     [ INFO ] GPU PCI id     : 1926
+     [ INFO ] GPU description: SKL ULT GT3
+     [ INFO ] no nomodeset in GRUB cmdline (good)
+     [ INFO ] Linux distro   : CentOS Linux 7.5.1804
+     [ INFO ] Linux kernel   : 5.0.3-1.el7.elrepo.x86_64
+     [ INFO ] glibc version  : 2.17
+     [ INFO ] gcc version    : 4.8.5-28) (>=4.8.2 suggested)
```
Comment (dvrogozh) [inline tools/sys_analyzer/README.md]: we recommend and kind of require gcc >=6 on a primary readme page...

File: tools/sys_analyzer/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    --------------------------
+    User and kernel mode drivers health checks:
+    --------------------------
+     [ WRN ] user not in the 'video' group. Add with 'usermod -a -G video {user}'
+     [ OK ] i915 driver in use by Intel video adapter
```
Comment (dvrogozh) [inline tools/sys_analyzer/README.md]: which driver is default (how LIBVA_DRIVER_NAME is setup)? what is the driver path?

File: tools/sys_analyzer/README.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+     [ OK ] libva pkg-config found
+     [ OK ] Intel iHD used by libva
+     [ OK ] vainfo reports valid codec entry points
+     [ WRN ] HuC not enabled. Make sure HuC firmware is installed
+     [ INFO ] Some HuC dependent features might be missing
+     [ OK ] libdrm check passed
+     [ OK ] /dev/dri/renderD128 connects to Intel i915
```
Comment (dvrogozh) [inline tools/sys_analyzer/README.md]: in multiadapters work we would need to explicitly list /dev/dri/card and /dev/dri/render matches for Intel? does tool support that now?

File: tools/sys_analyzer/sys_analyzer_linux.py [hunk 6]
```diff
@@ -0,0 +1,721 @@
+#!/usr/bin/python3
+"""
+
+Copyright (c) 2019 Intel Corporation All Rights Reserved.
+
+THESE MATERIALS ARE PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
```
Comment (dvrogozh) [inline tools/sys_analyzer/sys_analyzer_linux.py]: hmm... what is the license? I am not sure that this is MIT...

File: tools/sys_analyzer/sys_analyzer_linux.py [hunk 7]
```diff
@@ -0,0 +1,721 @@
...
+        output+=line
+    fin.close()
+    return output
+
+def parse_pciid(pciid):
+    sts=loglevelcode.ERROR
+    gpustr="unknown"
+
+    id_directory={
+    '0402':'HSW GT1 desktop',
```
Comment (dvrogozh) [inline tools/sys_analyzer/sys_analyzer_linux.py]: are there any available distro tools with this?

File: tools/sys_analyzer/sys_analyzer_linux.py [hunk 8]
```diff
@@ -0,0 +1,721 @@
...
+    '8A5A':'ICL GT1',
+    '8A5B':'ICL GT1',
+    '8A5C':'ICL GT1',
+    '8A5D':'ICL GT1',
+    '8A71':'ICL GT1',
+    '4500':'EHL GT2',
+    '4541':'EHL GT2',
+    '4551':'EHL GT2',
+    '4569':'EHL GT2',
+    '4571':'EHL GT2',}
```
Comment (dvrogozh) [inline tools/sys_analyzer/sys_analyzer_linux.py]: TGL?

**Comment by dvrogozh** [comment]
What do we want from this tool? To provide a status: "system is ready" and point to possible misconfiguration issues? Or we want it to be used to dump relevant system information and attach to bug for diagnosis?

I would suggest to consider both...

File: tools/sys_analyzer/README.md (hunk 3 above)
Comment (dmitryermilov) [reply tools/sys_analyzer/README.md]: This is old output

File: tools/sys_analyzer/README.md (hunk 5 above)
Comment (dmitryermilov) [reply tools/sys_analyzer/README.md]: Please check source code...

File: tools/sys_analyzer/README.md (hunk 4 above)
Comment (dmitryermilov) [reply tools/sys_analyzer/README.md]: Which driver path ? I don't follow your question.  Your highlighted something different.

**Comment by dmitryermilov** [comment]
> What do we want from this tool? To provide a status: "system is ready" and point to possible misconfiguration issues? Or we want it to be used to dump relevant system information and attach to bug for diagnosis?
> 
> I would suggest to consider both...

Both

File: tools/sys_analyzer/sys_analyzer_linux.py (hunk 8 above)
Comment (dmitryermilov) [reply tools/sys_analyzer/sys_analyzer_linux.py]: ok

File: tools/sys_analyzer/sys_analyzer_linux.py (hunk 7 above)
Comment (dmitryermilov) [reply tools/sys_analyzer/sys_analyzer_linux.py]: What?

File: tools/sys_analyzer/sys_analyzer_linux.py [hunk 9]
```diff
@@ -0,0 +1,743 @@
...
+  printf("%d.%d\\n",ver.Major,ver.Minor);
+  return 0;
+}
+"""
+
+    if pkgconfig.exists('libmfx'):
+        print_info("libmfx package found",loglevelcode.SUCCESS)
+    else:
+        print_info("libmfx package not found in default PKG_CONFIG locations",loglevelcode.WARNING)
+        os.environ["PKG_CONFIG_PATH"] = "/opt/intel/mediasdk/lib64/pkgconfig/"
```
Comment (dmitryermilov) [inline tools/sys_analyzer/sys_analyzer_linux.py]: need to check default msdk install path

File: tools/sys_analyzer/sys_analyzer_linux.py [hunk 10]
```diff
@@ -0,0 +1,743 @@
...
+
+    else:
+        print_info("Processor not Xeon or Core.",loglevelcode.INFO)
+
+    return loglevelcode.SUCCESS
+
+def is_OS_media_ready():
+
+    #check GPU PCIid
+    lspci_output=run_cmd("lspci -nn -s 0:02.0")
```
Comment (dmitryermilov) [inline tools/sys_analyzer/sys_analyzer_linux.py]: 0:02.0 ?

**Pull request closed without merging**